dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java

```diff
@@ -299,23 +297,27 @@ public String getDs3ObjectPath(final Path key, final Path value, final boolean c
     private void waitForPermanentStorageTransfer(final long totalJobSize) throws Exception {
         final boolean isCacheJobEnable = settings.getShowCachedJobSettings().getShowCachedJob();
         final String dateOfTransfer = DateFormat.formatDate(new Date());
+
+        updateProgress(totalJobSize, totalJobSize);
+        updateMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(),
+                FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer,
+                resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString());
+
+        loggingService.logMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(),
+                FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer,
+                resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString(), LogType.SUCCESS);
+
         if (isCacheJobEnable) {
-            updateProgress(totalJobSize, totalJobSize);
-            updateMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(), FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer, resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString());
-            updateProgress(totalJobSize, totalJobSize);
-            loggingService.logMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(), FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer, resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString(), LogType.SUCCESS);
-            GetJobSpectraS3Response response = ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId));
-            while (!response.getMasterObjectListResult().getStatus().toString().equals(StringConstants.JOB_COMPLETED)) {
+            while (!ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult().getStatus().toString().equals(StringConstants.JOB_COMPLETED)) {
                 Thread.sleep(60000);
```

Comment:
We should look at seeing if we can find a way to refactor this to remove the sleep.  In the sea vault code we used a recurring task to do something like this.  The main reason I'm not a fan of this kind of sleep is that we are consuming a whole thread while this is happening, rather than freeing the thread to do something else.